include/dlaf/common/index2d.tpp

```diff
@@ -60,15 +60,13 @@ Index2DType computeCoords(Ordering ordering, LinearIndexT index,
 ///
 /// @return -1 if given index is outside the grid size, otherwise the linear index (w.r.t specified ordering)
 /// @throws std::invalid_argument if ordering is not valid
-template <class Index2DType>
-typename Index2DType::IndexType computeLinearIndex(
-    Ordering ordering, const Index2DType& index,
-    const std::array<typename Index2DType::IndexType, 2>& dims) {
+template <class IndexType, class IndexTag>
+IndexType computeLinearIndex(Ordering ordering, const Index2D<IndexType, IndexTag>& index,
+                             const Size2D<IndexType, IndexTag>& dims) {
   if (!index.isIn(dims))
     return -1;
 
-  std::size_t ld_size_index = (ordering == Ordering::RowMajor) ? 1 : 0;
-  auto leading_size = dims[ld_size_index];
+  auto leading_size = (ordering == Ordering::RowMajor) ? dims.cols() : dims.rows();
```

Comment:
Relatively minor comment because the compiler probably optimizes is ti anyway but why not use `dims.cols()` and `dims.rows()` directly in the switch statement? Or just write this:

```
return (ordering == Ordering::ColumnMajor) ? dims.rows() * index.col() + index.row()
                                           : dims.cols() * index.row() + index.col();
```

Also, the method shouldn't throw, perhaps mark it with noexcept. I would also replace the `if (!index.isIn(dims))` check with DLAF_ASSERT_MODERATE or HEAVY.

Reply:
Good point! I used leading size because in the previous version was more clear like this, but now it is not necessary anymore!

About the `return -1` vs `DLAF_ASSERT_MODERATE`, it seems absolutely legit, but we should first agree on the API change.

@rasolca what do you think?

Reply:
> About the `return -1` vs `DLAF_ASSERT_MODERATE`, it seems absolutely legit, but we should first agree on the API change.

I don't mind, but you should check if the `-1` return was used somewhere.